src/main/java/org/reactome/addlinks/fileprocessors/HmdbMetabolitesFileProcessor.java

```diff
@@ -130,4 +129,54 @@ public HmdbMetabolitesFileProcessor()
 		return null;
 	}
 
+	
+	private void transformXmlToTsv(String inputFilename, String outfileName)
+	{
+		
+		try(OutputStream outStream = new FileOutputStream(outfileName);)
+		{
+			XMLInputFactory xif = XMLInputFactory.newInstance();
+			XMLStreamReader xsr = xif.createXMLStreamReader(new FileReader(inputFilename));
+			xsr.nextTag();
+			Source source = new StreamSource(this.getClass().getClassLoader().getResourceAsStream("hmdb_metabolites_transform.xsl"));
+			TransformerFactory factory = TransformerFactory.newInstance();
+			Transformer transformer = factory.newTransformer(source);
+			
+			while (xsr.nextTag() == XMLStreamConstants.START_ELEMENT)
+			{
+				if (xsr.getName().getLocalPart().equals(METABOLITE_ELEMENT_NAME))
+				{
+					StAXSource src = new StAXSource(xsr);
+					Result result = new StreamResult(outStream);
+					transformer.transform(src, result);
+				}
+			}
+			xsr.close();
+		}
+		catch (FileNotFoundException e ) 
+		{
+			logger.error("Input XML file was not found!", e);
+			e.printStackTrace();
+		}
+		catch (XMLStreamException e)
+		{
+			logger.error("Error streaming XML file!", e);
+			e.printStackTrace();	
+		}
+		catch (TransformerConfigurationException e)
+		{
+			logger.error("Error with XSL file!", e);
+			e.printStackTrace();
+		}
+		catch (TransformerException e)
+		{
+			logger.error("Data transformation error!", e);
+			e.printStackTrace();
+		}
+		catch (IOException e)
+		{
+			logger.error("I/O Error", e);
+			e.printStackTrace();
+		}
```

Comment:
Should a final general Exception try/catch be put at the end?

Reply:
Why? None of the methods being called throw `Exception`, they all throw more specific types of `Exception`.